instrumentation/grpc/src/main/java/brave/grpc/GrpcClientResponse.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.grpc;
+
+import brave.Response;
+import brave.Span;
+import brave.internal.Nullable;
+import io.grpc.ClientCall;
+import io.grpc.Metadata;
+import io.grpc.Status;
+
+// intentionally not yet public until we add tag parsing functionality
+final class GrpcClientResponse extends Response {
+  final GrpcClientRequest request;
+  @Nullable final Status status;
+  @Nullable final Metadata trailers;
+  @Nullable final Throwable error;
+
+  GrpcClientResponse(GrpcClientRequest request,
+    @Nullable Status status, @Nullable Metadata trailers, @Nullable Throwable error) {
```

Comment:
I think `status` or `trailers` are never `null`

Reply:
when we assume close is always called, I agree.. this ties to your other point, which I'm ok documenting in RATIONALE so in case the assumption proves wrong, we can update and move do this again

Just note if it does prove wrong we will have to mark something we said was not nullable nullable, which would surprise people.

Please ack if you feel confident enough that we should assume grpc always calls close, as that's not an assumption we make elsewhere, and frankly I would be surprised if that's provable.

Reply:
Thanks - after writing some tests, realized that indeed `close` isn't always called when interceptors are badly behaving.

For the nullability in particular, I noticed we only call `new Grpc*Response` in `close` though, elsewhere we just finish the span directly without using this class. I guess it makes sense, that a GrpcResponse is a gRPC response, where gRPC responses always have status / trailers as per the wire format, and any other case there is no response and this wouldn't be initialized, but maybe we're anticipating such a case so keeping it loose is fine too.

Reply:
appreciate the help with the tests as without them this is all folklore!

FWIW the code with "finish" is more forward thinking for #999 which would actually use the status stuff. more of this was re-jigging in anticipation of that being used... admittedly makes less sense until that it merged also.

Reply:
since the GrpcResponse types are hidden anyway, let's move this to #999 where you can see them used (and also where they are made public)